server/src/main/java/io/druid/guice/http/HttpClientModule.java

```diff
@@ -135,14 +138,23 @@ public HttpClient get()
     {
       final DruidHttpClientConfig config = getConfigProvider().get().get();
 
+      if (config.getUnusedConnectionTimeout() != null &&
```

Comment:
Probably better to check in DruidHttpClientConfig, so that every call site can benefit from the same check.

Reply:
Leaving check to the caller(HttpClientModule in this case) on how they want to interpret both the timeouts.

Reply:
I would agree here with @jihoonson to do it in DruidHttpClientConfig.getUnusedConnectionTimeout() so that callers don't need to bother with it.

Reply:
Thanks @himanshug @jihoonson, will move this check to `DruidHttpClientConfig`.